azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VirtualMachineScaleSetIpConfigurationProperties.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import java.util.List;
+
+@AutoValue
+public abstract class VirtualMachineScaleSetIpConfigurationProperties {
+
+   /**
+    * The public IP address configuration of the Virtual Machine Scale Set Ip Configuration Properties
+    */
+   @Nullable
+   public abstract VirtualMachineScaleSetPublicIPAddressConfiguration publicIPAddressConfiguration();
+
+   /**
+    * The subnet of the Virtual Machine Scale Set Ip Configuration Properties
+    */
+   public abstract Subnet subnet();
+
+   /**
+    * The private IP address version of the Virtual Machine Scale Set Ip Configuration Properties
+    */
+   @Nullable
+   public abstract String privateIPAddressVersion();
+
+   /**
+    * The load balancer backend address pools of the Virtual Machine Scale Set Ip Configuration Properties
+    */
+   @Nullable
+   public abstract List<IdReference> loadBalancerBackendAddressPools();
+
+   /**
+    * The load balancer inbound nat pools of the Virtual Machine Scale Set Ip Configuration Properties
+    */
+   @Nullable
```

Comment:
You are enforcing presence and immutability. Remove this annotation.

Reply:
These @Nullable annotations are required to serialized the json response from the service (especially the when the value is not returned as sometimes there are not setup by the users)

Reply:
Still needs to be removed

Reply:
Hi @nacx  I should have been more clear and place the mock error in the right comment section.  This is the file with the @nullable that I was referring when talking about the mocks failing.  I removed the others as you suggested, but the ones in this file seems they need to stay.

This is the error:
```testVMSSGet_StorageProfile(org.jclouds.azurecompute.arm.features.VirtualMachineScaleSetApiMockTest)  Time elapsed: 0.035 sec  <<< FAILURE!
org.jclouds.http.HttpResponseException: Error parsing input: Null loadBalancerInboundNatPools```

Reply:
Hi @nacx  I should have been more clear and place the mock error in the right comment section.  This is the file with the @nullable that I was referring when talking about the mocks failing.  I removed the others as you suggested, but the ones in this file seems they need to stay.

This is the error:
```testVMSSGet_StorageProfile(org.jclouds.azurecompute.arm.features.VirtualMachineScaleSetApiMockTest)  Time elapsed: 0.035 sec  <<< FAILURE!
org.jclouds.http.HttpResponseException: Error parsing input: Null loadBalancerInboundNatPools```